File: sfn/await/SfnIsComplete.cs
namespace sfn_ut.sfn.await;

public abstract class SfnIsComplete<PAYLOAD> : SfnStep {
  public abstract SfnIsCompleteOutput<PAYLOAD> IsComplete(PAYLOAD payload);

  public SfnIsComplete() : base(StepType.IsComplete) {
    // Do nothing
  }
}

File: java-sfn-unit-test/src/main/java/com/sfnut/sfn/lambdawithloop/SfnIsComplete.java
package com.sfnut.sfn.lambdawithloop;

import com.sfnut.sfn.SfnStep;
import com.sfnut.sfn.StepType;

public abstract class SfnIsComplete<PAYLOAD> extends SfnStep {
    
    public SfnIsComplete() {
        super(StepType.LAMBDA_LOOP_IS_COMPLETE);
    }

    public abstract SfnIsCompleteOutput<PAYLOAD> isComplete(PAYLOAD payload);
}
